source/retryable-writes/tests/README.rst

```diff
@@ -395,10 +395,67 @@ and sharded clusters.
 
    9. Disable the failpoint.
 
+#. Test that drivers return the original error after encountering a
+   WriteConcernError error with a RetryableWriteError label. This test MUST be
+   implemented by any driver that implements the Command Monitoring
+   specification. This prose test requires drivers to set a fail point after an
+   ``insertOne`` operation but before the subsequent retry. Drivers that are
+   unable to set a failCommand after the CommandSucceededEvent SHOULD use
+   mocking or write a unit test to cover the same sequence of events.
+
+   1. Create a client with ``retryableWrites=true``. If testing against a
```

Comment:
I see, thank you for that thorough explanation.

Reply:
Yes mongos would need to apply the same rule as in this PR. Otherwise I believe it could return a non-retryable NoWritesPerformed-like error to the driver even though the write actually did succeed. @prestonvasquez could you open a server ticket for mongos to implement this change?

Reply:
@ShaneHarvey @visemet [SERVER-69295](https://jira.mongodb.org/browse/SERVER-69295)